Add localization support for server messages

def- opened the pull request:
by translating them on client-side.

Texts containing %s/%d are hard this way, replaced some, didn't localize some others. But I guess even incomplete this is better than nothing.

The reason not to do it serverside is because you might have to translate to different language for each player.

def- committed:
Add localization support for server messages

by translating them on client-side.

Texts containing %s/%d are hard this way, replaced some, didn't localize
some others. But I guess even incomplete this is better than nothing.

The reason not to do it serverside is because you might have to
translate to different language for each player.

def- committed:
Update data/languages files

fokkonaut commented:
Should be done serverside. Server should pop up a voting window when joining a server asking to translate to your language based on IP Address, hitting F3 accepts. Otherwise language should be choosable by /language en/de/ru/tr/idk

def- committed:
Add German translation for server messages

def- commented:
Oh no, I thought about this too simply. It doesn't actually work for replacing parts of texts and doing so will be quite hacky.

pure-luck-999 commented:
Consider using terms not English strings as keys. Like `game.mod.infcroya.class.biologist.title`

Rationale behind this is English strings will be changed too (typos, renames and so on) but you will have to use old string as keys forever.

def- commented:
Yes, that's clearly better, but I'm not planning to clean up the entire code base in this change :)

def- committed:
Fix up the server translation support

Still missing:

- scripts/languages needs to create ~~ instead of == if the file is in
  server
- lots of strings with Console()->Print(...) left to Localize

I'm not sure if I like this PR so much anymore. Thoughts?

def- commented:
Fix up the server translation support

Still missing:

- scripts/languages needs to create ~~ instead of == if the file is in
  server
- lots of strings with Console()->Print(...) left to Localize

I'm not sure if I like this PR so much anymore. Thoughts?

fokkonaut commented on data/languages/german.txt:
```diff
@@ -1213,289 +1213,295 @@ Client message
 
 Server does not allow voting to move players to spectators
-== Server erlaubt keine Abstimmungen um Spiler zu Beobachtern zu verschieben
+~~ Server erlaubt keine Abstimmungen um Spiler zu Beobachtern zu verschieben
```
Spiler xd

heinrich5991 commented:
I might take a look at it to do it without sacrificing the ability to change English strings.
